### PR TITLE
Extract daemon statsCollector to its own package

### PR DESCRIPTION
**Pull request by vdemeester**
This moves `statsCollector` to a `stats` package and update the daemon to work with it. Should be easier to test and has its own responsablity not depending on the daemon (with the `supervisor` interface it did not really anyway).

I think I didn't break Solaris (/cc @amitkris)

/cc @dnephin @thaJeztah @icecrime @cpuguy83 @AkihiroSuda @tiborvass @tonistiigi 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Review by dnephin — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
Shall we try to get https://github.com/docker/docker/pull/25361 merged first? That's been open for such a long time

**Comment by ripcurld0** [comment]
LGTM 🇫🇷

**Review by boaz0 — COMMENTED** [review]
LGTM :fr:

File: daemon/stats/collector_solaris.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package stats
```
Comment (LK4D4) [inline daemon/stats/collector_solaris.go]: Solaris :crying_cat_face:

**Review by LK4D4 — APPROVED** [review]
LGTM

**Pull request merged**